From 4b75f5f32add4586bdc42e2ae602f5074634fe69 Mon Sep 17 00:00:00 2001
Date: Wed, 16 Dec 2015 17:30:59 +0200
Subject: [PATCH 11/25] apply backport SmPL patch
 collateral-evolutions/network/0035-skb_no_fcs/skb_no_fcs.cocci

---
 drivers/net/ethernet/intel/igb/igb_main.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index ae5e871..84e2b5e 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -4812,7 +4812,10 @@ static u32 igb_tx_cmd_type(struct sk_buff *skb, u32 tx_flags)
 				 (E1000_ADVTXD_MAC_TSTAMP));
 
 	/* insert frame checksum */
-	cmd_type ^= IGB_SET_FLAG(skb->no_fcs, 1, E1000_ADVTXD_DCMD_IFCS);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
+	cmd_type ^= IGB_SET_FLAG(skb->no_fcs, 1, E1000_ADVTXD_DCMD_IFCS)
+#endif /* if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0) */
+;
 
 	return cmd_type;
 }
-- 
2.6.3

